### PR TITLE
[SERVER-1146] Vault only migration

### PR DESCRIPTION
**Pull request by nathanfishcircleci**
split up the functions into files and then created a dedicated vault only migration

unfortunately I did not test this and needs to be tested

### CODE REVIEW

File: migrate/2.19-export-vault.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+KEY_BU="circle-data"
+VAULT_BU="circleci-vault"
+
+##
+# Main function
+##
+function circleci_database_export() {
+    echo "Starting CircleCI Database Export"
+
+    vault-preflight_checks
```
Comment (ryanwohara) [inline migrate/2.19-export-vault.sh:53]:
```suggestion
    vault_preflight_checks
```

File: migrate/2.19-export-vault.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+##
+# import individual scripts/functions
+##
+
+source 2.19-preflight.sh
+source 2.19-postgres.sh
+source 2.19-mongo.sh
+source 2.19-vault.sh
```
Comment (ryanwohara) [inline migrate/2.19-export-vault.sh:10]: These are only fine locally - we don't presently scp these to the 2.x host.

File: migrate/3.0-vault.sh [hunk 3]
```diff
@@ -0,0 +1,156 @@
+#!/bin/bash
+
+##
+# CircleCI Database Import Script
```
Comment (ryanwohara) [inline migrate/3.0-vault.sh:4]:
```suggestion
# CircleCI Vault & Key Import Script
```

File: migrate/3.0-vault.sh [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        echo "Namespace '$NAMESPACE' not found."
+        exit 1
+    elif [ ! -s $PG_BU/circle.sql ]
+    then
+        echo "Postgres data at '$PG_BU/circle.sql' not found (or is empty)"
+        exit 1
+    elif [[ $(du -sm $MONGO_BU 2>/dev/null | awk '{print $1}') -lt 2 ]] # If the size is under 2MB something went wrong
+    then
+        echo "Mongo data at '$MONGO_BU' not found (or is empty)"
+        exit 1
```
Comment (ryanwohara) [inline migrate/3.0-vault.sh:50]: These won't be present in the vault import

File: migrate/3.0-vault.sh [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    MONGO_POD="mongodb-0"
+    MONGODB_USERNAME="root"
+    MONGODB_PASSWORD=$(kubectl -n $NAMESPACE get secrets mongodb -o jsonpath="{.data.mongodb-root-password}" | base64 --decode)
+
+    kubectl -n $NAMESPACE exec $MONGO_POD -- mkdir -p /tmp/backups/
+    kubectl -n $NAMESPACE cp -v=2 $MONGO_BU $MONGO_POD:/tmp/backups/
+
+    kubectl -n $NAMESPACE exec $MONGO_POD -- mongorestore --drop -u $MONGODB_USERNAME -p $MONGODB_PASSWORD --authenticationDatabase admin /tmp/backups/circleci-mongo-export/;
+    kubectl -n $NAMESPACE exec $MONGO_POD -- rm -rf /tmp/backups
+}
```
Comment (ryanwohara) [inline migrate/3.0-vault.sh:94]: These will fail in the Vault import

File: migrate/3.0-vault.sh [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+    preflight_checks
+
+    echo "...scaling application deployments to 0..."
+    scale_deployments 0
+
+    # wait one minute for pods to scale down
+    sleep 60
+
+    import_postgres
+    import_mongo
```
Comment (ryanwohara) [inline migrate/3.0-vault.sh:143]:
```suggestion
```

File: migrate/3.0-vault.sh (hunk 4 above)
Comment (ryanwohara) [reply migrate/3.0-vault.sh:50]:
```suggestion
```

File: migrate/3.0-vault.sh (hunk 5 above)
Comment (ryanwohara) [reply migrate/3.0-vault.sh:94]:
```suggestion
```

File: migrate/3.0-vault.sh [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+    echo ""
+    echo "###################################"
+    echo "##   Encryption & Signing keys   ##"
+    echo "##   must be uploaded to kots.   ##"
+    echo "###################################"
+    echo ""
+    echo "You may find your keys here:  $(pwd)/${KEY_BU}"
+
+    ls -lh $KEY_BU | grep -v ^total
+}
```
[the remark this answers is not in the document]
Comment (ryanwohara) [reply migrate/3.0-vault.sh:129]:
```suggestion
source 3.0-bottoken.sh
source 3.0-key.sh
```

**Review by ryanwohara — APPROVED** [review]
LGTM :)

**Review by nsmith5 — APPROVED** [review]

**Pull request merged**